### ISSUE TITLE
macOS 10.15.3 Catalina - OBS is not requesting Screen Recording permission, unable to do window capture

### ISSUE THREAD

**Issue by ctsrc** [obsproject/obs-studio#2455]
## Expected Behavior

According to https://obsproject.com/forum/threads/macos-10-15-catalina-support-status.111343/ OBS Studio version 24.0.6 is able to work on macOS 10.15 Catalina.

## Current Behavior

When I add a Window Capture source in OBS, only OBS windows themselves, and the desktop background and the menubar show up.

<img width="728" alt="window_capture" src="https://user-images.githubusercontent.com/36199671/75892609-e95fce80-5e31-11ea-81b8-5ae3b3891328.png">

If I tick the box below the window list that says:

- [ ] Show windows with empty names

Then the other windows show up in the list with the name of the application followed by "(null)".

<img width="729" alt="null" src="https://user-images.githubusercontent.com/36199671/75892987-7a36aa00-5e32-11ea-978b-f507338a9394.png">

Selecting any of those yields no actual capture.

I also went into the Security & Privacy settings on my computer.

OBS is not present at all in the list of applications that want permission to perform screen recording.

<img width="674" alt="screen_recording" src="https://user-images.githubusercontent.com/36199671/75893638-6ccdef80-5e33-11ea-9d40-f16cee3b3dd1.png">

Only in the Microphone and the Camera permission lists is OBS present.

<img width="674" alt="mic" src="https://user-images.githubusercontent.com/36199671/75894201-2b8a0f80-5e34-11ea-9498-448eb634be2b.png">

<img width="675" alt="cam" src="https://user-images.githubusercontent.com/36199671/75894213-304ec380-5e34-11ea-8c7a-6bffce8e9e48.png">

The result of this is that I can capture video from my webcam, and audio from my mic, but I can not capture any windows (other than OBS itself, and the macOS menu bar and the desktop wallpaper).

## Steps to Reproduce

1. Installed OBS 24.0.6 on macOS Catalina 10.15.3, running on a MacBook Air 2018 model.
2. Ran OBS.
3. Allowed OBS to capture video and sound.
4. Tried to capture window. Results in behavior described above.

## Additional information

Latest version of OBS at the time of this writing (OBS 24.0.6) was installed via Homebrew.

My MacBook Air 2018 model has an external monitor connected to it. I find it unlikely that this has anything to do with the problem but it's worth mentioning anyway I think.

**Comment by ctsrc** [comment]
Aaaaand immidiatelly after submitting this issue I figured out what the problem was.

So here's what happened after I had submitted the issue: I began closing down applications in order to restart the machine to see if that would help.

On my external monitor I had a game running in full screen, which I was trying to record.

When I exited said game, I saw that additional macOS permission popups were on that display.

Whereas the permission dialogs for microphone and for camera had popped up on the built-in display on the MacBook Air, macOS had "wisely" decided to put the permission dialogs for Screen Recording and some other stuff on the external monitor where I could not see them due to the game that was currently running in full screen on that display.

Of course it had to be something stupid like that 😅.

Closing this issue now. Hopefully this issue submission ends up being useful to someone else that experiences this too in the future 😂.
